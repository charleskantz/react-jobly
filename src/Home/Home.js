import React, { useContext } from 'react';
import AuthContext from '../AuthContext';
import { HomeHeading, HomeSubHeading } from '../Common/Type';
import { ButtonLink } from '../Common/StyledLink';
import { HomeDiv, Div } from '../Common/Div';
import heroImg from '../Common/home_hero_img.svg';
import { Image } from '../Common/Image';

// Homepage Component
function Home() {

  const { userInfo } = useContext(AuthContext);

  const buttonHref = userInfo ? '/companies' : '/login';

  return (
    <HomeDiv display="flex">
      <Image src={heroImg} size="280"/>
      <Div margin="1rem">
        {userInfo &&
          <HomeSubHeading>
            Welcome back, {userInfo.first_name}.
          </HomeSubHeading>
        }
        <HomeHeading>Apply to the best jobs with Jobly</HomeHeading>
        <HomeSubHeading>
          Find and apply privately with one-click applications.
          See salary and equity upfront.
        </HomeSubHeading>
        <ButtonLink solid href={buttonHref} >Find Jobs Now</ButtonLink>
      </Div>
    </HomeDiv>
  );
}

export default Home;