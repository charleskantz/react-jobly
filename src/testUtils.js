import React from "react";
import AuthContext from "./AuthContext";

// mock of context for testing purposes

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
  jobs: ['jobone', 'jobtwo', 'jobthree']
};

const AuthProvider = ({ children, userInfo = demoUser }) => (
  <AuthContext.Provider value={{userInfo}}>
    {children}
  </AuthContext.Provider>
);

export { AuthProvider };
