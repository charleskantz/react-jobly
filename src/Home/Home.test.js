import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Home from "./Home";
import { AuthProvider } from "../testUtils";

//smoke
it("renders", function () {
  render(
    <MemoryRouter>
      <AuthProvider>
        <Home />
      </AuthProvider>
    </MemoryRouter>
  )
});

// snapshots
it("matches snapshot when logged in", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <AuthProvider>
        <Home />
      </AuthProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <AuthProvider userInfo={null}>
        <Home />
      </AuthProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

// button link
it("links you to /companies when logged in", function () {
  const { getByText } = render(
    <MemoryRouter>
      <AuthProvider>
        <Home />
      </AuthProvider>
    </MemoryRouter>
  );

  const button = getByText('Find Jobs Now');
  expect(button.getAttribute('href')).toBe('/companies');
})

it("links you to /login when logged out", function () {
  const { getByText } = render(
    <MemoryRouter>
      <AuthProvider userInfo={null} >
        <Home />
      </AuthProvider>
    </MemoryRouter>
  );

  const button = getByText('Find Jobs Now');
  expect(button.getAttribute('href')).toBe('/login');
})
