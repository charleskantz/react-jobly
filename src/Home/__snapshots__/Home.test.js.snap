// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <div
    class="css-ioutv6"
    display="flex"
  >
    <img
      class="css-106gtww"
      size="280"
      src="home_hero_img.svg"
    />
    <div
      class="css-1numv7f"
    >
      <h1
        class="css-14exbet"
      >
        Welcome back, testfirst.
      </h1>
      <h1
        class="css-cyue6k"
      >
        Apply to the best jobs with Jobly
      </h1>
      <h1
        class="css-14exbet"
      >
        Find and apply privately with one-click applications. See salary and equity upfront.
      </h1>
      <a
        class="css-1dyiqdp"
        href="/companies"
      >
        Find Jobs Now
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="css-ioutv6"
    display="flex"
  >
    <img
      class="css-106gtww"
      size="280"
      src="home_hero_img.svg"
    />
    <div
      class="css-1numv7f"
    >
      <h1
        class="css-cyue6k"
      >
        Apply to the best jobs with Jobly
      </h1>
      <h1
        class="css-14exbet"
      >
        Find and apply privately with one-click applications. See salary and equity upfront.
      </h1>
      <a
        class="css-1dyiqdp"
        href="/login"
      >
        Find Jobs Now
      </a>
    </div>
  </div>
</DocumentFragment>
`;
